network/p2p/gossip/gossipable.go

```diff
@@ -29,3 +29,20 @@ type Set[T Gossipable] interface {
 	// corresponding salt.
 	GetFilter() (bloom []byte, salt []byte)
 }
+
+type AddManySet[T Gossipable] interface {
```

Comment:
I know this would increase the scope of this change (especially because changing interfaces that are depended on by Coreth creates annoy circular dependencies)... But should we just be updating the `Set` interface to require this rather than adding this optional interface?

It seems like the X-chain and the P-chain could also try and avoid grabbing their locks multiple times.

Reply:
Sure, that was exactly my concern, trying to add this feature in a way without modifying the existing interface. But if it's OK, then I can just refactor `Add` to take a slice instead of a single element.

Reply:
I think we should probably push for just having one `Add`. Let me know if you have any questions on how to handle this weird circular import.

Reply:
+1. Having an interface that has both `Add` and the `AddMany` signature is going to be an awkward API IMO. I would recommend an API that uses a variadic parameter like:

```
Add(gossipables ... Gossipable) []error
```